extide/gradle/src/org/netbeans/modules/gradle/api/GradleBaseProjectBuilder.java

```diff
@@ -125,6 +127,9 @@ void processTasks() {
 
     void processDependencies() {
 
+        File gradleUserHome = (File) info.get("gradle_user_home");
```

Comment:
Here the `gradle_user_home` value is read...

Reply:
Technically it is an API, a protocol behind a serialization between the netbeans-gradle-tooling which is actually a trojan horse that lets serialize the build information out of Gradle. On NetBeans side this data is de-serialized by the different GradleWhateverProjectBuilders. These builders then constructs public API objects from the de-serialized information.

This is one of the ugliest part of my Gradle integration. There are two reasons for implementing it this way. The ready to use Gradle Models were really not provided enough information in the days of Gradle 1.x, 2.x and 3.x series. The other one is that the remoting which would make these rich interfaces easily consumable were/are memory grubbing proxies. So I reduced the remoting to HashSet-s, HashMaps with Strings and Files.
Having that once NetBeans get the information proxies back, basically a few hashmaps, it rewrites the data into real JVM objects, so those can be serialized to disk for caching, and also can built API objects out of it. AFAIK, Attila had a nice abstraction between Gradle and NetBeans and the objects (proxies) returned by  Gradle were ready to use for NetBeans. Unfortunately they were seldom reclaimed by the GC. I guess Attila fixed that later, but I have not followed his plugin that closely.